K2Bridge/KustoDAL/KustoResponseParser.cs

```diff
@@ -189,10 +194,20 @@ private void ReportQueryExecutionMetrics(KustoResponseDataSet kustoResponseDataS
                 {
                     foreach (DataRow row in parsedKustoResponse[AggregationTableName].TableData.Rows)
                     {
-                        IBucket bucket = createBucketFromDataRow(row);
-                        response.AddBucketToAggregation(bucket);
+                        Bucket bucket = createBucketFromDataRow(row);
+                        if (bucket != null)
+                        {
+                            response.AddBucketToAggregation(bucket);
+                        }
                     }
                 }
+
+                // For Range aggregations, the calculated total hits is wrong, so we have an additional column with the expected count
```

Comment:
Can you explain further? why is it wrong?

Reply:
In general, for aggregations, we determine the total hit count by adding the buckets doc counts together. However, this does not work for range buckets *with overlapping ranges*: in this case our total hit count is wrong (or at least, different from what ES returns) because we then count docs twice. ES returns the total hit count over the query but without double counting. I added this just so that we can get the same total hit count as ES.